livedatagraph.cpp

```diff
@@ -1,51 +1,101 @@
 #include "livedatagraph.h"
 
+/*!
+  \class LiveDataGraph
+  \brief The LiveDataGraph class displays live data coming from a TCP server in real time
+
+  Live data graphs take a TCP socket as a pointer and displays the data coming from the server in real time.
+
+  It handles:
+  1. Adding the graph to the UI
+  2. Changing the display depending on the attribute wanted (gear, launch, etc)
+  3. Ability to delete the graph
+ */
 LiveDataGraph::LiveDataGraph(SocketConnection * socket, QObject *parent) : QObject(parent)
 {
+    // Basic setup of the live graph
+    this->socket = socket;
     this->graph =  new QCustomPlot;
     this->graph->addGraph();
     this->graph->graph(0)->rescaleAxes();
     this->graph->update();
     this->displayAttribute = "Select Attribute";
 
-    //add title to graph
+    // Add title to graph
     QCPTextElement *graphTitle = new QCPTextElement(this->graph, "Live Graph", QFont("sans", 17, QFont::Bold));
     this->graph->plotLayout()->insertRow(0);
     this->graph->plotLayout()->addElement(0, 0, graphTitle);
 
-    //adds dropdown element box
-    QStringList commands = { "Select Attribute", "Speed", "RPM", "Breaking" };
+    // Adds dropdown element box
+    QStringList commands = { "Select_Attribute", "Gear", "Launch", "Traction", "Autoshift", "RPM", "Voltage", "F_Press", "O_Temp", "C_Temp", "O_Press" };
```

Comment:
Can this be dynamic at run-time based on the keys in the map?

Reply:
hmmm I'll be honest I'm not too familiar with dynamic run-time programming with C++, I'm not sure